core/src/main/java/feign/stream/StreamDecoder.java

```diff
@@ -43,27 +45,36 @@
  *   Stream<Contributor> contributors(@Param("owner") String owner, @Param("repo") String repo);
  * }</code>
  * </pre>
+ * 
+ * @author mroccyen
  */
 public final class StreamDecoder implements Decoder {
 
   private final Decoder iteratorDecoder;
 
-  StreamDecoder(Decoder iteratorDecoder) {
+  public StreamDecoder(Decoder iteratorDecoder) {
```

Comment:
just to make myself very clear, I'm not against your idea of having 2 decoders, I think that was really smart and adds tons of flexibility that this class don't have right now.

My only "issue" was changing the signature for `public static StreamDecoder create(Decoder iteratorDecoder) {`

The create, with single arg, needs to take the `iteratorDecoder` not the delegate decoder.

What if we did something like this:

```
private final Decoder iteratorDecoder;
private final Optional<Decoder> delegateDecoder;
```

```
if (!isStream(type)) {
  if(delegateDecoder.isEmpty()) {
    throw new IllegalArgumentException("StreamDecoder supports only stream: unknown " + type);
  } else {
    return delegateDecoder.get().decode(response, type);
  }
}
```

Reply:
If there is only one constructor parameter like iteratorDecoder, then how can the deleteDecoder parameter be passed to StreamDecoder, via the setter method?

Reply:
You can have multiple `create()` methods.... I just want the original one to keep current functionality

since constructor is not public, you could change it to have 2 arguments and have one create with empty and other with a delegate decoder.

Reply:
Thanks, I will modify it like this.